packages/shared/scope-hoisting/src/link.js

```diff
@@ -337,32 +337,28 @@ export function link({
           }
         } else {
           if (mod.meta.id && assets.get(assertString(mod.meta.id))) {
+            let name = assertString(mod.meta.exportsIdentifier);
             // Replace with nothing if the require call's result is not used.
             if (!isUnusedValue(path)) {
-              let name = assertString(mod.meta.exportsIdentifier);
               node = t.identifier(replacements.get(name) || name);
+            }
+
+            // Insert __esModule interop flag if the required module is an ES6 module with a default export.
+            // This ensures that code generated by Babel and other tools works properly.
+            if (mod.meta.isES6Module && mod.symbols.has('default')) {
+              let binding = path.scope.getBinding(name);
+              if (binding && !binding.path.getData('hasESModuleFlag')) {
+                if (binding.path.node.init) {
+                  binding.path
+                    .getStatementParent()
+                    .insertAfter(ESMODULE_TEMPLATE({EXPORTS: name}));
+                }
 
-              // Insert __esModule interop flag if the required module is an ES6 module with a default export.
-              // This ensures that code generated by Babel and other tools works properly.
-              if (
-                asset.meta.isCommonJS &&
```

Comment:
Was there a reason why `isCommonJS` was checked here? From my point of view, it does not matter as long as it is an es6 module.